src/Markup/Avalonia.Markup/Markup/Parsers/Nodes/StringIndexerNode.cs

```diff
@@ -29,15 +29,15 @@ protected override bool SetTargetValueCore(object value, BindingPriority priorit
             var list = target as IList;
             var dictionary = target as IDictionary;
             var indexerProperty = GetIndexer(typeInfo);
-            var indexerParameters = indexerProperty?.GetIndexParameters();
+            ParameterInfo[] indexerParameters;
```

Comment:
Why move this initialization into the `if`? It's slightly less readable like this to me, though I don't really mind either way, just wondering what the reason is.

Reply:
IIRC, it had a warning because `indexerParameters` could be `null` (in `indexerParameters.Length == Arguments.Count`), although we know it's not, because it's only `null` when `indexerProperty` is `null` as well, but the compiler doesn't know that. I also agree that it's slightly less readable.